cmd/crane/cmd/root.go

```diff
@@ -146,3 +152,27 @@ func (ht *headerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
 	}
 	return ht.inner.RoundTrip(in)
 }
+
+type warnCollector struct {
+	warns map[string]struct{}
+}
+
+func (d *warnCollector) Add(warning string) {
+	d.warns[warning] = struct{}{}
+}
+
+func (d *warnCollector) Report() {
+	warns := make([]string, 0, len(d.warns))
+	for k := range d.warns {
+		warns = append(warns, k)
+	}
+	sort.Strings(warns)
+	prefix := "\033[1;33m[WARNING]\033[0m:"
+	if v, set := os.LookupEnv("NO_COLOR"); set && v != "0" {
```

Comment:
there's also various detection approaches to try to see if the writer is a terminal that we think can display color output - xref https://github.com/kubernetes/kubernetes/blob/fe91bc257b505eb6057eb50b9c550a7c63e9fb91/staging/src/k8s.io/kubectl/pkg/util/term/term.go - it might make sense to only enable color output automatically when we think we're in an interactive terminal

Reply:
Thanks, this was helpful. I omitted the `github.com/moby/term.IsTerminal` call for now mainly to avoid the dependency. We might add it later, it looks pretty small but it seems to bring in `github.com/Azure/go-ansiterm` for some reason too.